src/transforms/deployment.ts

```diff
@@ -0,0 +1,113 @@
+import { Context } from "probot/lib/context";
+import issueKeyParser from "jira-issue-key-parser";
+
+// We need to map the state of a GitHub deployment back to a valid deployment state in Jira.
+// https://docs.github.com/en/rest/reference/repos#list-deployments
+// Deployment state - GitHub: Can be one of error, failure, pending, in_progress, queued, or success
+// https://developer.atlassian.com/cloud/jira/software/rest/api-group-builds/#api-deployments-0-1-bulk-post
+// Deployment state - Jira: Can be one of unknown, pending, in_progress, cancelled, failed, rolled_back, successful
+function mapState(state: string): string {
+	switch (state) {
+		case "queued":
+		case "pending":
+			return "pending";
+		case "in_progress":
+			return "in_progress";
+		case "success":
+			return "successful";
+		case "error":
+		case "failure":
+			return "failed";
+		default:
+			return "unknown";
+	}
+}
```

Comment:
Don't know why the tabbing looks messed here but it looks good on my IDE
![Screen Shot 2021-08-09 at 1 20 21 pm](https://user-images.githubusercontent.com/58801956/128657198-3c3e13e3-94b9-40f6-8102-5f7394c48ce8.png)

Reply:
it's fine :)
It's because it's using tabs and for some reason github shows tabs as 8 spaces, which is massive...